src/pudl/output/ferc1.py

```diff
@@ -1912,12 +1941,78 @@ def forest(self: Self) -> nx.DiGraph:
         if pruned_nodes := set(nodes_before_pruning).difference(nodes_after_pruning):
             raise AssertionError(f"Unexpectedly pruned stepchildren: {pruned_nodes=}")
 
-        # forest = self.set_forest_attributes(
-        #    forest,
-        #    exploded_meta=self.exploded_meta,
-        #    exploded_calcs=self.exploded_calcs,
-        #    tags=self.tags,
-        # )
+        # HACK alter.
+        # two different parents. those parents have different sets of dimensions.
+        # sharing some but not all of their children so they weren't caught from in the
+        # only stepchildren node removal from above. a generalization here would be good
+        almost_pure_stepparents = [
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "depreciation_amortization_and_depletion_utility_plant_leased_to_others",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "depreciation_and_amortization_utility_plant_held_for_future_use",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "utility_plant_in_service_classified_and_unclassified",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+        ]
+        forest.remove_nodes_from(almost_pure_stepparents)
+
+        # Ensure that we haven't removed any calculation components that *would* have
+        # altered the final root-to-leaf calculations.
+        # * Weights pertain to child nodes.
+        # * But the nodes we're removing above are guaranteed to be parents (Though
+        #   many them will also be children)
+        # * We want to check that any of them that *are* children have a weight of 1.0
```

Comment:
I realized that a lot of the complexity in all this was because the weights were being associated with nodes, rather than edges -- they really are an attribute of a calculation which links two nodes... and if we put the weight on the edge then there's no problem with having lots of different weights associated with different edges that point at the same node.  And by virtue of enforcing a tree structure, we necessarily end up with a single remaining incoming edge for each node, with a single weight, removing this potential conflict / ambiguity.